doc/editing 03 development process (#623)
Together-100Devs/Together: docs/04-Contributing.md

   - Review [Testing Docs Page](https://github.com/Together-100Devs/Together/tree/main/docs/06-Testing) for more information on how testing works.

> 7. Rebase your work before pushing.

Reviewer: maybe we want to link to the other section where you wrote this out? I'm not sure, open to suggestions. It does seem like we will have to maintain two somewhat identical descriptions in the docs and make sure they also stay in sync with each other

Reviewer 2: Yeah I had a similar thought to that, I ended up making page 4 a shorter version of 3. 3 and 4 are tied into eachother so there's no a clear way to split them up. I'll think on it!

Reviewer 2: Ideas for the future:

1. Keep them both separate but adjust where the "editing code and submitting a pull request logic goes" to avoid duplicated logic:

change 3 ==> how to contribute (with editing code and submitting a pull request section cut out)
change 4 ==> development process (+ the editing code and submitting a pull request section)

Since the contribution naturally flows into the development process?

2. Merge them both into the same file
page 4 (contributing)  is mostly the "Editing code and submitting a pull request" section, so it'd be very small with that cut out

3. Just keep it as it is, and deal with duplicated logic. Focus on keeping the version on page 4 a shorter version of 3

I'd lean towards sticking with 3 for this PR, but in the future make a future PR if we do decide to change up the flow?

   `git fetch upstream main` : get the latest updates from the original repo's main branch.

   `git rebase upstream/main` : rebase your feature branch on top of the latest upstream/main changes. Upstream is now the new "base" for your feature branch to sit on top of,

>    Then use interactive rebase to squash your commits into a single commit:

Reviewer: ```suggestion
8.  Squash your commits into a single commit with interactive rebase:
```

   `git rebase -i upstream/main`: view all the commits in your current branch, since you branched off of upstream/main (oldest first).

   You want to change all but the first from pick to squash:

>    > pick c32423434 first commit
>    > squash af243df second commit

Reviewer: ```suggestion
   > pick c32423434 first commit  
   > squash af243df second commit
```

   Save and close the editor, you'll be prompted to edit the final commit message (which starts off as all of the previous messages combined).

   optional: `git log --oneline`: if you want to double check that it merged into one commit. Or on upper left side of VSCode click on source control to see the history (the icon with 3 circles, connected by 2 lines)

> 8. Push the changes to your fork on GitHub.

Reviewer: ```suggestion
9. Push the changes to your fork on GitHub.
```
